spec/unit/etc_spec.rb

```diff
@@ -0,0 +1,155 @@
+require 'puppet/etc'
+require 'spec_helper'
+
+# The Ruby::Etc module is largely non-functional on Windows - many methods
+# simply return nil regardless of input, the Etc::Group struct is not defined,
+# and Etc::Passwd is missing fields
+describe Puppet::Etc, :if => !Puppet.features.microsoft_windows? do
```

Comment:
Per the preceding code comment, the upstream (ruby) Etc module methods return nil on Windows, and the `Etc::Passwd` and `Etc::Group` structs are missing member fields, so I've gated this test to not run on Windows. The original code that used Etc in Puppet appears to all be posix-specific and Windows is thusly avoided which makes sense. I haven't gated loading of `Puppet::Etc` above behind some sort of Windows platform test, but if one does `require 'puppet/etc'` on Windows and uses the methods of `Puppet::Etc`, they'll just return `nil` like their upstream counterparts. That made the most sense to me at the time as I was trying to mimic upstream behavior as closely as possible (excepting the UTF-8 conversion) but I'm all ears if there's a more standard way to do this.